### DISCUSSION TITLE
Extend the search feature in the ID field

### DISCUSSION THREAD

**Discussion by ProvaTeams** [rustdesk/rustdesk#5350] [Feature Request]
When typing an ID in the Control Remote Desktop field, it would be useful to enable the search for recent sessions for that ID, as in the Search ID field.

**Comment by rustdesk** [comment]
@sahilyeole

**Comment by beelchester** [comment]
I have finished this feature but got stuck on a small and embarassing thing.
@21pages or @fufesou can you please help me on this?

I'm using the Autocomplete widget for this feature. It has `fieldViewBuilder` I have made it just like the previous Id field but I got stuck on managing `TextEditingController` and also maybe `FocusNode`. The fieldViewBuilder requires its own `TextEditingController` and `FocusNode` so it conflicts with already working `_idController` and `_idFocusNode`. If I use one the other's logic doesn't work. I tried a few things but didn't work, not sure how to do this.

My code: https://github.com/rustdesk/rustdesk/compare/master...sahilyeole:rustdesk:feat/autocomplete

Also, If there are any UI suggestions, I can make the changes.
Working for all the required fields a peer has, and searching recent, fav, lan, address book, group peers.

https://github.com/rustdesk/rustdesk/assets/73148455/d478d470-924e-4cee-86fb-5fa80f418ce3

**Reply by 21pages** [reply]
try RawAutocomplete

**Comment by rustdesk** [comment]
https://github.com/rustdesk/rustdesk/pull/6040
